Simple Incremental/Assets/Scripts/CharacterHealthEffect.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(CharacterHealth))]
+public class CharacterHealthEffect : MonoBehaviour
+{
+    CharacterHealth characterHealth;
+    ParticleSystem system;
+
+    void Awake()
+    {
+        characterHealth = GetComponent<CharacterHealth>();
+        system = GetComponent<ParticleSystem>();
+    }
+    void Start()
+    {
+        characterHealth.HealthChanged += playHitEffect;
+    }
+
+    private void playHitEffect()
+    {
+        //When players health changes play damage partice effect
+        if (system != null)
+        {
+            var main = system.main;
+            main.startLifetime = 0.5f;
```

Comment:
Not sure but maybe you could move magic numbers to constants somewhere?

Reply:
Set the number as a configurable field. [SerializeField] a private variable so it's configurable via inspector!

Reply:
Yeah, i feel like there is a better way to do this but I need to research how. Do people create seperate particle systems per effect than select them based on name. The reason I say that is there is hundreds of things to change in the particle system. I don't think people just have one than tweak the settings to chage it but, not really sure. Mine was just a real simple one as a test mainly. Ill remove those values they shouldn't really be needed if the system isn't getting modified.

Reply:
Yeah, i feel like there is a better way to do this but I need to research how. Do people create seperate particle systems per effect than select them based on name. The reason I say that is there is hundreds of things to change in the particle system. I don't think people just have one than tweak the settings to chage it but, not really sure. Mine was just a real simple one as a test mainly. Ill remove those values they shouldn't really be needed if the system isn't getting modified.

Reply:
People often just define the particle system separately and invoke .Play() from custom scripts.